### PR TITLE
Fix pagination load more

### LINKED ISSUE #12491: Broken Order List Pagination
**Describe the bug**
The order list sometime refreshes and resets the scrolling position when scrolling down. Moreover, it seems that afterwards it sometimes stops loading more pages.

**To Reproduce**
1. Clear apps data
2. Login
3. Open Order List and scroll down
4. Observe that a PTR animation (the small circle indicator at the top) appears at certain point.
5. Wait until it completes -> notice the scrolling position resets back to 0.

**Screenshots**

https://github.com/user-attachments/assets/155c495f-852d-497d-8340-d68cd87f38fc
https://github.com/user-attachments/assets/eece001f-8a60-40bf-9575-c5a23618b79f

**Expected behavior**
The pagination should keep loading more pages until I reach the last order and shouldn't reset my scrolling position.

**Isolating the problem (mark completed items with an [x]):**
- [x] I can reproduce this bug consistently using the steps above.

**Mobile Environment**
Please include:
- Device: Pixel 8
- Android version: Android 14 and 15
- WooCommerce Android version: 10.1.1

### PR DESCRIPTION
**Pull request by atorresveiga**
Closes: #12491

### Description
This PR fixes the issue that prevented the load more behavior from triggering after reaching the end of the screen. The problem was that because of a timeout on the `getListState` function, the function returned a `need-to-refresh` state instead of the required to trigger the load more behavior `can-load-more`. 

This PR fixes this issue with two main changes:

1. It updates the FluxC version to use a new implementation that will not expire the final states `can-load-more` and `fetched`.
2. Update the `ShouldUpdateOrdersList` to consider the list status. With this change, we aim to prevent inconsistent states in case the app is closed before reaching one of the final states.

### Steps to reproduce
1. Open the app
2. Navigate to the orders screen
3. Pull-to-refresh the screen
4. Check that the last update timestamp is updated
5. Cose the app and wait 2 min (more than 1 min will work too)
6. Re-open the app 
7. Navigate to the orders screen
8. Scroll to the bottom of the screen
9. Check that the load more behavior is not triggered and only the first page is displayed

### Testing information
1. Open the app
2. Navigate to the orders screen
3. Pull-to-refresh the screen
4. Check that the last update timestamp is updated
5. Cose the app and wait 2 min (more than 1 min will work too)
6. Re-open the app 
7. Navigate to the orders screen
8. Scroll to the bottom of the screen
9. Check that the pagination is working as expected

### The tests that have been performed

1. Test that pagination works as expected when data is loaded from a cache older than 1 min.
2. Test that receiving an order notification updates the list
3. Test that creating a new order displays the new order in the list
4. Test that applying a status filter is working as expected
5. Test that completing an order in the list with the "processing" filter removes the order from the list.

### Images/gif

https://github.com/user-attachments/assets/8d2a4b44-97eb-467d-ab7a-9b1582dc0c1e

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/234">20.3</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/list/ShouldUpdateOrdersList.kt [hunk 1]
```diff
@@ -2,12 +2,21 @@ package com.woocommerce.android.ui.orders.list
...
 import kotlinx.coroutines.flow.first
+import org.wordpress.android.fluxc.model.list.ListDescriptor
+import org.wordpress.android.fluxc.model.list.ListState
+import org.wordpress.android.fluxc.store.ListStore
 import javax.inject.Inject
 
-class ShouldUpdateOrdersList @Inject constructor(private val lastUpdateDataStore: LastUpdateDataStore) {
-    suspend operator fun invoke(listId: Int): Boolean {
-        return lastUpdateDataStore.getLastUpdateKeyByOrdersListId(listId).let { key ->
+class ShouldUpdateOrdersList @Inject constructor(
```
Comment (malinajirka) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/list/ShouldUpdateOrdersList.kt:10]: @atorresveiga I'm wondering if we truly need this, it feels a bit off or at least unexpected to me. One of the main goals of the ListStore component is to manage lists for us automatically and avoid handling refreshes from the client. The `ListState.NEEDS_REFRESH` is an event emitted and consumed automatically within fluxC which should take care of updating the list for us and propagating that to the UI. We shouldn't need to check this state here and act on it. Could you please elaborate on why we need to handle this manually here?

P.S. I read the PR description, but I'm either misunderstanding it or it doesn't explain why a client side fix is needed.

Comment (atorresveiga) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/list/ShouldUpdateOrdersList.kt:10]: Hi @malinajirka 👋   thank for sharing your concerns.

> One of the main goals of the ListStore component is to manage lists for us automatically and avoid handling refreshes from the client.

While the ListStore component could provide many handy tools to implement pagination and the common logic to fetch fresh data, I feel that it should be up to the client apps to decide whether this data is stale. This is what happened with the Background Tasks project.

The Background Task project aimed to improve the app's perceived performance by relying more on cached data. As part of this project, we wanted to decide whether the cached data was outdated. For this, we defined a 30-minute threshold. If the cached data is not older than 30 minutes, we present that data; otherwise, we fetch the info from the API.

>  Could you please elaborate on why we need to handle this manually here?

Sure! As mentioned above, we wanted to check if the data in the DB was older than 30 minutes and decide if we needed to re-fetch everything or rely on that data cached. The decision of whether the data is stale or not is now owned by Woo. With this goal in mind, I implemented the `ShouldUpdateOrdersList`. This use case checks if the last update timestamp for the list description is still valid and returns a boolean that we use later to fetch the order's info or not. 

Now, why do I need to check the `ListState.NEEDS_REFRESH` status?
In FluxC, we have implemented an expired logic for list states, you can check the function `isListStateOutdated`. 

Here is the description for that function:

> A helper function that returns whether it has been more than a certain time has passed since it's lastModified.
Since we keep the state in the DB, in the case of application being closed during a fetch, it'll carry over to the next session. To prevent such cases, we use a timeout approach. If it has been more than a certain time since the list is last updated, we should ignore the state.

So, there could be edge cases where the app is closed when the list is in an intermediary state (like `fetching-frist-page`). In those scenarios, I would want to refresh everything even if I have a valid timestamp. Does that make sense? 
To sum up, I want the last update timestamp to be the source of truth when deciding whether to update the list or not, unless the list is in an inconsistent state. Let me know if this information helps! It's important to mention that as part of the [FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/3090), I prevented the final states `can_load_more` and `fetched` from expiring. This means we can still use them even if they are older than 1 minute, and because they are final states, I don't think there was a reason to discard them.
